agent/src/agentclient.rs

```diff
@@ -354,10 +382,18 @@ impl<T: APIServerClient> BrupopAgent<T> {
                 }
                 BottlerocketShadowState::MonitoringUpdate => {
                     event!(Level::INFO, "Monitoring node's healthy condition");
-                    self.uncordon().await?;
+                    self.handle_recover().await?;
                     // TODO: we need add some criterias here by which we decide to transition
                     // from MonitoringUpdate to WaitingForUpdate.
                 }
+                BottlerocketShadowState::ErrorReset => {
+                    // Spec state should never be ErrorReset
+                    event!(Level::ERROR, "Spec state should never be ErrorReset");
+                    return agentclient_error::Assertion {
+                        message: "Spec state should never be ErrorReset.".to_string(),
+                    }
+                    .fail();
+                }
             }
         } else {
             event!(Level::INFO, "Did not detect action demand.");
```

Comment:
I wonder if this log message should be `DEBUG` level? I'm realizing this might be quite noisy since it would likely be emitted on most runs through the event loop.